### ISSUE TITLE
Generate shell completion scripts

### ISSUE THREAD

**Issue by claudiodangelis** [claudiodangelis/qrcp#128]
`cobra`, the library used to implement qrcp's command line, supports the generation of shell completion scripts for bash and zsh (and more?).

We should add the ability to generate such scripts for qrcp: https://github.com/spf13/cobra/blob/master/bash_completions.md

These scripts should also be included in the deb and rpm packages, as well as in the .tar.gz packages.

**Comment by shilangyu** [comment]
I would like to propose a different approach: Personally I don't like when the completions are bundled with the release. Looking for the files is often cumbersome.

As an example, [Deno](https://github.com/denoland/deno) provides a subcommand to generate them. Example: `deno completions fish > /my/fish/dir/deno.fish`. This ensures I don't have to look for the completions files, they can be generated at runtime.

**Comment by claudiodangelis** [comment]
Thanks for the suggestion @shilangyu.

**Comment by mathieu-aubin** [comment]
so does `npm`. Very curious tho... aren't completion scripts generally in a pretty much universal/standard path (for linux anyways) from distro to distro? Also, why would you want or need to 'look for the file' once it's properly installed, within the proper folder and most likely, loaded upon session/shell creation?

Dont get me wrong, i am not trying to say you are wrong or doing something you shouldn't-- i am genuinely interested/curious as to why one would need to worry about files that dont contain (as far as i am concerned) any user-friendly or have no value for a user as a file.

Thanks for helping me understand!

**Comment by shilangyu** [comment]
I think you misunderstood me @mathieu-aubin (if you were replying to my comment). I install/update qrcp with `go`. Meaning I don't dabble in the release artifacts. So for me it's much easier to:

```sh
$ go get github.com/claudiodangelis/qrcp
$ qrcp completions fish > /my/fish/dir/qrcp.fish # the CLI might have new commands, I need to update the completions
```

vs.

```sh
$ go get github.com/claudiodangelis/qrcp
$ curl https://github.com/claudiodangelis/qrcp/path/to/release/file.tar.gz
$ # unzipping file.tar.gz ...
$ mv file/completions.fish /my/fish/dir/qrcp.fish
$ rm file
```

**Comment by mathieu-aubin** [comment]
Which makes sense, totally, yes! I had blinds while focusing on releasers and packaging and yes, forgot that most still do install using go so fetching for a file is unpractical, at least alot more than being able to self-generate. Gotcha! Thanks for kickstarting my cafeine!

**Comment by jmoscoso83** [comment]
Sorry someone is hacking my accounts, so I dont know what this people are
doing behind me.

**Comment by claudiodangelis** [comment]
@shilangyu after some thinking I don't see a real benefit in having the ability to generate shell completion scripts implemented as a command, mostly because:
- it is not be a qrcp functionality
- it would only be executed once

Shell completion scripts can be part of the repository itself, not only part of the release. The workflow could be something like:

```sh
go get github.com/claudiodangelis/qrcp
curl https://raw.githubusercontent.com/claudiodangelis/qrcp/master/extra/completion/qrcp.fish -O /my/fish/qrcp.fish
```


Makes sense?

**Comment by shilangyu** [comment]
@claudiodangelis fair enough. Either way, looking forward for the completions :)

**Comment by claudiodangelis** [comment]
Hello @shilangyu, I had a few second thoughts and decided to add the `completion` command as you suggested.
Would you like to give it a try? It's on the feat-shell-completion-scripts branch at the moment.

go get github.com/claudiodangelis/qrcp@feat-shell-completion-scripts

    qrcp completion fish | source

Thank you!
C

**Comment by shilangyu** [comment]
Seems to be working great! 

![image](https://user-images.githubusercontent.com/29288116/92018358-974a2800-ed44-11ea-9b7f-b81f5d89eb17.png)

![image](https://user-images.githubusercontent.com/29288116/92018869-471f9580-ed45-11ea-9958-1807fb281805.png)

**Comment by eggeek** [comment]
Most of the case, we just need `qrcp send <file>` or `qrcp recive <dir>`, 
It could be better to add `'(-)*:file or directory:_files'` in `_qrcp_send` and `_qrcp_recive` completion, so that we can complete file/dir name.

**Comment by zhangnew** [comment]
> Most of the case, we just need `qrcp send <file>` or `qrcp recive <dir>`, It could be better to add `'(-)*:file or directory:_files'` in `_qrcp_send` and `_qrcp_recive` completion, so that we can complete file/dir name.

This feature has not been implemented as of version [0.10.1](https://github.com/claudiodangelis/qrcp/tree/0.10.1)
